examples/weather/shared/src/location/model/geocoding_response.rs

```diff
@@ -0,0 +1,167 @@
+            display_option(&self.ar),
+            display_option(&self.ascii),
+            display_option(&self.az),
+            display_option(&self.bg),
+            display_option(&self.ca),
+            display_option(&self.da),
+            display_option(&self.de),
+            display_option(&self.el),
+            display_option(&self.en),
+            display_option(&self.eu),
+            display_option(&self.fa),
+            display_option(&self.feature_name),
+            display_option(&self.fi),
+            display_option(&self.fr),
+            display_option(&self.gl),
+            display_option(&self.he),
+            display_option(&self.hi),
+            display_option(&self.hr),
+            display_option(&self.hu),
+            display_option(&self.id),
+            display_option(&self.it),
+            display_option(&self.ja),
+            display_option(&self.la),
+            display_option(&self.lt),
+            display_option(&self.mk),
+            display_option(&self.nl),
+            display_option(&self.no),
+            display_option(&self.pl),
+            display_option(&self.pt),
+            display_option(&self.ro),
+            display_option(&self.ru),
+            display_option(&self.sk),
+            display_option(&self.sl),
+            display_option(&self.sr),
+            display_option(&self.th),
+            display_option(&self.tr),
+            display_option(&self.vi),
+            display_option(&self.zu),
+        )
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialOrd, PartialEq, Default, Clone)]
+pub struct GeocodingResponse {
+    pub name: String,
+    pub local_names: Option<LocalNames>,
+    pub lat: f64,
+    pub lon: f64,
+    pub country: String,
+    pub state: Option<String>,
+}
+
+impl fmt::Display for GeocodingResponse {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(
+            f,
+            "GeocodingResponse: (name: {}, local_names: {}, lat: {}, lon: {}, country: {}, state: {})",
+            self.name,
+            display_option(&self.local_names),
+            self.lat,
+            self.lon,
+            self.country,
+            display_option(&self.state),
+        )
+    }
+}
+
+pub static SAMPLE_GEOCODING_RESPONSE: LazyLock<Vec<GeocodingResponse>> = LazyLock::new(|| {
+    vec![GeocodingResponse {
+        name: "Phoenix".to_string(),
+        local_names: None,
+        lat: 33.456_789,
+        lon: -112.037_222,
+        country: "US".to_string(),
+        state: Some("Arizona".to_string()),
+    }]
+});
+
+pub static SAMPLE_GEOCODING_RESPONSE_JSON: LazyLock<String> =
+    LazyLock::new(|| serde_json::to_string(&*SAMPLE_GEOCODING_RESPONSE).unwrap());
+
+#[derive(Debug, Serialize, Deserialize, PartialOrd, PartialEq, Default, Clone)]
+pub struct ZipCodeResponse {
+    pub zip: String,
+    pub name: String,
+    pub lat: f64,
+    pub lon: f64,
+    pub country: String,
+}
```

Comment:
Might be worth putting those into JSON tests fixtures.  They look like they are only used in tests anyway.